eisbuk-admin/src/components/atoms/OrganizationSettings/OrganizationSettings.tsx

```diff
@@ -0,0 +1,397 @@
+    if (enableEdit) resetForm();
+    setEnableEdit(!enableEdit);
+  };
+
+  const emailFields: Partial<FieldProps>[] = [
+    {
+      name: "emailNameFrom",
+      label: "Email Name From",
+    },
+    {
+      name: "emailFrom",
+      label: "Email From",
+    },
+    {
+      name: "emailTemplate",
+      label: "Email Template",
+    },
+  ];
+  const smsFields: Partial<FieldProps>[] = [
+    {
+      name: "smsTemplate",
+      label: "SMS Template",
+    },
+  ];
+
+  const currentUser = userAuthInfo?.email || "";
+  return (
+    <>
+      <div className={classes.title}>
+        <Typography variant="h4">{`${
+          organization.organizationName || "Organization"
+        }  Settings`}</Typography>
+      </div>
+      <div className={classes.content}>
+        <Formik
+          {...{ initialValues: { ...organization } }}
+          onSubmit={handleSubmit}
+          validateOnChange={false}
+          validationSchema={OrganizationValidation}
+        >
+          {({ errors, isSubmitting, isValidating, values, resetForm }) => (
+            <Form className={classes.form}>
+              <FormControl component="fieldset">
+                <h5 className={classes.sectionTitle}>Name</h5>
+                <Field
+                  key="organizationName"
+                  label="Organization Name"
+                  name="organizationName"
+                  className={classes.field}
+                  as={TextField}
+                  aria-label="Organization Name"
+                  variant={enableEdit ? "outlined" : "filled"}
+                  disabled={!enableEdit}
+                  value={values.organizationName || ""}
+                />
+
+                <Divider />
+                <h5 className={classes.sectionTitle}>Admins</h5>
+                <AdminsField
+                  enableEdit={enableEdit}
+                  currentUser={currentUser}
+                />
+                <Divider />
+                <h5 className={classes.sectionTitle}>Email</h5>
+                <div className={classes.fieldSection}>
+                  {emailFields.map((field) => (
+                    <Field
+                      key={field.name}
+                      label={field.label}
+                      name={field.name}
+                      className={
+                        field.name?.includes("Template")
+                          ? classes.templateField
+                          : classes.field
+                      }
+                      as={TextField}
+                      aria-label={field.label}
+                      variant={enableEdit ? "outlined" : "filled"}
+                      disabled={!enableEdit}
+                      multiline={
+                        field.name?.includes("Template") ? true : false
+                      }
+                      rows={field.name?.includes("Template") ? "4" : "1"}
+                      value={values[`${field.name}`] || ""}
+                    />
+                  ))}
```

Comment:
There are two identical `Array.map()` calls: for sms and email fields. It would be better (for code redundancy and readability) to create a `renderTextfields` function (or whatever you want to name it) which would accept a parameter of `typeof emailFields` and return mapped fields (a simple copy paste of the selected code should do)

Reply:
Additionally, I still stand by my proposal that it would be best to make this into a full component, rather than a function.